androguard/misc.py

```diff
@@ -40,8 +40,11 @@ def AnalyzeAPK(filename, session=None):
     if not session:
         session = get_default_session()
 
-    with open(filename, "rb") as fd:
-        data = fd.read()
+    if raw:
+        data = filename
+    else:
+        with open(filename, "rb") as fd:
+            data = fd.read()
 
     session.add(filename, data)
```

Comment:
if you supply raw data, the filename will be set with the raw data... this is probably not good...
maybe calcuate a hash over the raw data first and use it as filename? dunno..

otherwise it looks good!